modules/core/04-channel/keeper/packet.go

```diff
@@ -63,17 +63,12 @@ func (k Keeper) SendPacket(
 		return 0, errorsmod.Wrap(connectiontypes.ErrConnectionNotFound, channel.ConnectionHops[0])
 	}
 
-	clientState, found := k.clientKeeper.GetClientState(ctx, connectionEnd.ClientId)
-	if !found {
-		return 0, clienttypes.ErrClientNotFound
-	}
-
 	// prevent accidental sends with clients that cannot be updated
 	if status := k.clientKeeper.GetClientStatus(ctx, connectionEnd.ClientId); status != exported.Active {
 		return 0, errorsmod.Wrapf(clienttypes.ErrClientNotActive, "cannot send packet using client (%s) with status %s", connectionEnd.ClientId, status)
 	}
 
-	latestHeight := clientState.GetLatestHeight()
+	latestHeight := k.clientKeeper.GetLatestHeight(ctx, connectionEnd.ClientId)
```

Comment:
probably should do a `latestHeight.IsZero` check here, otherwise the timeout height is disregarded:
```go
// heightElapsed returns true if the timeout height is non empty
// and the timeout height is greater than or equal to the relative height.
func (t Timeout) heightElapsed(height clienttypes.Height) bool {
    return !t.Height.IsZero() && height.GTE(t.Height)
}
```

Reply:
I'm not sure I follow this one, this `latestHeight` var is pased to `timeout.Elapsed` which performs the check you've just showed.

Reply:
Cool, so check `latestHeight.IsZero` and return error before creating the `timeout`?